workers/unity/Packages/io.improbable.gdk.core/Tests/Editmode/Utility/CommandLineUtilityTest.cs

```diff
@@ -66,5 +67,71 @@ public void ParseCommandLineArgs_should_process_args_when_string_array_is_provid
             Assert.AreEqual("first-value", args.GetCommandLineValue("key1", "not-value"));
             Assert.AreEqual("second-value", args.GetCommandLineValue("key2", "not-value"));
         }
+
+        [Test]
+        public void ParseCommandLineArgs_can_parse_flag_arguments()
+        {
+            var args = CommandLineArgs.From(new List<string>
+            {
+                "+key1",
+                "first-value",
+                "+flag",
+                "+key2",
+                "second-value"
+            });
+            Assert.True(args.Contains("key1"));
+            Assert.True(args.Contains("key2"));
+            Assert.True(args.Contains("flag"));
+        }
+
+        [Test]
+        public void ParseCommandLineArgs_can_parse_flag_arguments_at_end_of_arg_list()
+        {
+            var args = CommandLineArgs.From(new List<string>
+            {
+                "+key1",
+                "first-value",
+                "+key2",
+                "second-value",
+                "+flag"
+            });
+            Assert.True(args.Contains("key1"));
+            Assert.True(args.Contains("key2"));
+            Assert.True(args.Contains("flag"));
+        }
+
+        [Test]
+        public void TryGetCommandLineValue_throws_when_accessing_flag_only_args()
+        {
+            var args = CommandLineArgs.From(new List<string>
+            {
+                "+key1",
+                "first-value",
+                "+key2",
+                "second-value",
+                "+flag"
+            });
+
+            var boolVal = false;
+            Assert.Throws<InvalidOperationException>(() => args.TryGetCommandLineValue("flag", ref boolVal));
+            var stringVal = string.Empty;
+            Assert.Throws<InvalidOperationException>(() => args.TryGetCommandLineValue("flag", ref stringVal));
+        }
+
+        [Test]
+        public void TryGetCommandLineValue_is_valid_for_empty_arg()
+        {
+            var args = CommandLineArgs.From(new List<string>
+            {
+                "+key1",
```

Comment:
This would be a better test by changing the keys around. An empty one first, and then one with a value.
This way you test that the parser doesn't skip the empty key